src/software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic_test.py

```diff
@@ -0,0 +1,364 @@
+    "blue_bots, yellow_bots, ball_initial_pos, ball_initial_velocity",
+    [
+        # Test left and right align to centre crease
+        (
+            tbots_cpp.Field.createSSLDivisionBField()
+            .friendlyDefenseArea()
+            .posXPosYCorner()
+            + tbots_cpp.Vector(0.5, 0),
+            tbots_cpp.Point(1, 0),
+            tbots_cpp.Point(0, 0),
+            tbots_cpp.Vector(0, 0),
+        ),
+        (
+            tbots_cpp.Field.createSSLDivisionBField()
+            .friendlyDefenseArea()
+            .posXNegYCorner()
+            + tbots_cpp.Vector(0.5, 0),
+            tbots_cpp.Point(1, 0),
+            tbots_cpp.Point(0, 0),
+            tbots_cpp.Vector(0, 0),
+        ),
+    ],
+)
+def test_crease_positioning(
+    blue_bots,
+    yellow_bots,
+    ball_initial_pos,
+    ball_initial_velocity,
+    simulated_test_runner,
+):
+
+    # Setup Robot
+    def setup(*args):
+        simulated_test_runner.simulator_proto_unix_io.send_proto(
+            WorldState,
+            create_world_state(
+                blue_robot_locations=[blue_bots],
+                yellow_robot_locations=[yellow_bots],
+                ball_location=ball_initial_pos,
+                ball_velocity=ball_initial_velocity,
+            ),
+        )
+
+        # These aren't necessary for this test, but this is just an example
+        # of how to send commands to the simulator.
+        #
+        # NOTE: The gamecontroller responses are automatically handled by
+        # the gamecontroller context manager class
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.STOP, team=Team.UNKNOWN
+        )
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.FORCE_START, team=Team.BLUE
+        )
+
+        # Setup Tactic
+        params = AssignedTacticPlayControlParams()
+
+        params.assigned_tactics[0].crease_defender.CopyFrom(
+            CreaseDefenderTactic(
+                enemy_threat_origin=tbots_cpp.createPointProto(ball_initial_pos),
+                crease_defender_alignment=2,
+                max_allowed_speed_mode=MaxAllowedSpeedMode.PHYSICAL_LIMIT,
+            )
+        )
+
+        simulated_test_runner.blue_full_system_proto_unix_io.send_proto(
+            AssignedTacticPlayControlParams, params
+        )
+
+        # Setup no tactics on the enemy side
+        params = AssignedTacticPlayControlParams()
+        simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+            AssignedTacticPlayControlParams, params
+        )
+
+    # Always Validation
+    always_validation_sequence_set = [
+        [
+            RobotNeverEntersRegion(
+                regions=[
+                    tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea(),
+                    tbots_cpp.Field.createSSLDivisionBField().enemyHalf(),
+                    tbots_cpp.Circle(ball_initial_pos, 1),
```

Comment:
Curious why the robot should never get close to `ball_initial_pos`? Are we assuming that the ball is always placed far away?

Reply:
Ah okay, so this test strictly assesses the positioning logic of the robot and that it defaults to a region around the friendly defence area, regardless of the initial position of the robot. The later tests then further test the crease's chipping, stealing, and blocking behaviour.

Reply:
That makes sense. But do we need to have the `tbots_cpp.Circle(ball_initial_pos, 1)` region as well?

Reply:
Yes, because this allows us to ensure that our DribbleFSM behaviour is not erroneously stealing the ball when it most definitely should not be (it is right next to an enemy).